core/src/main/java/org/apache/iceberg/avro/BuildAvroProjection.java

```diff
@@ -106,10 +106,16 @@ public Schema record(Schema record, List<String> names, Iterable<Schema.Field> s
             "Missing required field: %s", field.name());
         // Create a field that will be defaulted to null. We assign a unique suffix to the field
         // to make sure that even if records in the file have the field it is not projected.
+        String origFieldName = field.name();
+        boolean isValidFieldName = AvroSchemaUtil.validAvroName(origFieldName);
+        String fieldName =  isValidFieldName ? origFieldName : AvroSchemaUtil.sanitize(origFieldName);
```

Comment:
extra space after '='

Reply:
Also, I think isValidFieldName can be inline.

Reply:
Updated.